### PR TITLE
*(dm): support MySQL 8.0 collations

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/4948

### What is changed and how it works?

as title

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release branch (?)

### Release note 

```release-note
support upstream MySQL 8.0 collations
```

### CODE REVIEW

**Comment by lance6716** [comment]
at least this test always fail

```
[2022-03-23T07:24:30.179Z]     server_test.go:943: 
[2022-03-23T07:24:30.179Z]         	Error Trace:	server_test.go:943
[2022-03-23T07:24:30.179Z]         	Error:      	Received unexpected error:
[2022-03-23T07:24:30.179Z]         	            	EOF
[2022-03-23T07:24:30.179Z]         	Test:       	TestServerDataLossAfterUnregisterHandle
```

**Review by liuzix — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: dm/pkg/schema/tracker.go [hunk 1]
```diff
@@ -201,6 +202,8 @@ func NewTracker(ctx context.Context, task string, sessionCfg map[string]string,
 			return nil, err
 		}
 	}
+	// skip DDL test https://github.com/pingcap/tidb/pull/33079
+	se.SetValue(sessionctx.QueryString, "skip")
```
Comment (Ehco1996) [inline dm/pkg/schema/tracker.go:206]: just a question： do we need set this var in every func? can we just set once when `tracker` init

**Review by Ehco1996 — APPROVED** [review]
LGTM

File: dm/pkg/schema/tracker.go (hunk 1 above)
Comment (lance6716) [reply dm/pkg/schema/tracker.go:206]: tracker.Exec will write this field, and later DDL API will see outdated value.

**Pull request merged**